src/connection.js

```diff
@@ -126,6 +128,10 @@ class Connection extends EventEmitter {
         this.config.options.database = config.options.database;
       }
 
+      if (config.options.datefirst != undefined) {
+        this.config.options.datefirst = config.options.datefirst;
```

Comment:
Please add a range check here. You can follow the pattern for 'port' config option.
Please add a test. You can follow the pattern for 'badPort' test.
Sorry for not catching this sooner. Thanks.